pkg/controller/common/common.go

```diff
@@ -123,18 +123,46 @@ func (controllerContext ControllerContext) Stop() {
 	controllerContext.Broadcaster.Shutdown()
 }
 
-type ChildMap map[string]map[string]*unstructured.Unstructured
+type GroupVersionKind struct {
+	schema.GroupVersionKind
+}
+
+func (gvk GroupVersionKind) MarshalText() ([]byte, error) {
+	var marshalledText string
+	if gvk.Group == "" {
+		marshalledText = fmt.Sprintf("%s.%s", gvk.Kind, gvk.Version)
+	} else {
+		marshalledText = fmt.Sprintf("%s.%s/%s", gvk.Kind, gvk.Group, gvk.Version)
+	}
+	return []byte(marshalledText), nil
+}
+
+func (gvk *GroupVersionKind) UnmarshalText(text []byte) error {
+	kindGroupVersionString := string(text)
+	parts := strings.SplitN(kindGroupVersionString, ".", 2)
+	groupVersion, err := schema.ParseGroupVersion(parts[1])
+	if err != nil {
+		return err
+	}
+	*gvk = GroupVersionKind{
```

Comment:
This seems to be difficult to use from the point of client code.
This does not look idiomatic.
Have you thought of GVKUtil struct that exposes Unmarshal & Marshal methods?

Reply:
I am unsure if it works with `encoding.json` unmarshall method, from its apidoc map key should inmplement `TextUnmarshaller`.  Also, I must note that only place where this is needed are integration tests